### PR TITLE
Add ability to run fixed or interval forever

### PR DESCRIPTION
**Pull request by hrosenbe**
This pull request includes the following changes:
- Added a parameter called `runForever` which, if set to true, causes a fixed or interval runStrategy run to continue forever until manually terminated.
- Added a parameter `runDuration` that controls the length of a run when using the interval runStrategy.  This replaces the use of rampUp, numQosPeriods, etc to determine the length of an interval run.
- Moved the interpretation of the `repeatUserLoadPath` parameter from the run harness to the workload driver.  This was necessary in order to be able to support runForever.
- Changed the default value of `repeatUserLoadPath` to true.
- Added documentation for the interval runStrategy and other new parameters.
- Added documentation for how to access the browser interface of the auction application.

### CODE REVIEW

File: doc/userDocs/usersGuide.md [hunk 1]
```diff
@@ -607,6 +609,141 @@ edit the configuration file as follows:
...
+users incremented every 10 seconds. In this case, the increment will be (4000 -
+2000) / (120 / 10) = 166.667. Note that the driver will adjust this number so
+that some intervals increase by 166 and some by 167 so that the total increase
+is 2000 users.
+
+The third interval demonstrates that the "startUsers" parameter is optional. If
+omitted, the interval will start with the same number of users as were active at
+the end of the previous interval. If "startUsers" is omitted from the first
+interval, then the interval will start with 0 users. This particular example
+will ramp down from 4000 to 1000 users over 120 seconds.
```
Comment (bhoflich) [inline doc/userDocs/usersGuide.md:695]: over 240 seconds?

File: doc/userDocs/usersGuide.md [hunk 2]
```diff
@@ -1736,6 +1884,100 @@ configuration file you should:
...
+
+1. Get the EXTERNAL-IP address for the one of the nodes of your Kubernetes
+cluster using the command `kubectl get node -o wide`.  You can use the address
+of any node. Here we will assume that that the value is 192.168.20.20. 
+1. Get external port number assigned to port 443 on the nginx service using the
+command `kubectl get svc -n auctionw1i1`. The port number you want will be the
+right-hand half of a pair that looks like `443:30204/TCP`, where we have used
+30204 as an example.
+1. In a browser which can access the node EXTERNAL-IP address, enter the URL
+`https://192/168.20.20:30204`, where you should replace the IP address and port
```
Comment (bhoflich) [inline doc/userDocs/usersGuide.md:1941]: 192/ -> 192.

File: doc/userDocs/usersGuide.md (hunk 1 above)
Comment (hrosenbe) [reply doc/userDocs/usersGuide.md]: fixed

File: doc/userDocs/usersGuide.md (hunk 2 above)
Comment (hrosenbe) [reply doc/userDocs/usersGuide.md]: fixed

File: doc/userDocs/usersGuide.md [hunk 3]
```diff
@@ -607,6 +609,141 @@ edit the configuration file as follows:
...
+default value for this parameter is 1800.
+1. `userLoadPath` : This specifies a list of intervals of fixed duration. Within
+each interval the number of users either ramps up or down between start and end
+values, or takes a fixed value. 
+
+During a run with the `interval` runStrategy, the load driven to the application
+instances will change according to the intervals specified in the userLoadPath.
+The intervals are used in the order in which they are specified.  If the
+`runDuration` is longer than the sum of the durations of the intervals, then the
+userLoadPath will continue from the first interval until the run completes.  If
```
[the remark this answers is not in the document]
Comment (hrosenbe) [reply doc/userDocs/usersGuide.md]: Thanks.  I have made this change

**Pull request merged**